activerecord/lib/active_record/counter_cache.rb

```diff
@@ -105,7 +105,8 @@ def update_counters(id, counters)
         end
 
         if touch
-          updates << sanitize_sql_for_assignment(touch_updates(touch))
+          touch_updates = touch_updates(touch)
+          updates << sanitize_sql_for_assignment(touch_updates) unless touch_updates.empty?
```

Comment:
if the last item of `updates` is an empty string,  ActiveRecord generates an SQL like below.

```ruby
updates = ["\"credit_limit\" = COALESCE(\"credit_limit\", 0) + 1", ""]
unscoped.where(primary_key => id).update_all updates.join(", ")
# => UPDATE "accounts" SET "credit_limit" = COALESCE("credit_limit", 0) + 1,  WHERE "accounts"."id" = ?
```

This causes `ActiveRecord::StatementInvalid`.